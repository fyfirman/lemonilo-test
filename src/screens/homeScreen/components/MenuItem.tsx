import React from 'react';
import styled from 'styled-components/native';
import Icon from 'react-native-ionicons';
import myTheme from '@theme/theme';

const Container = styled.TouchableOpacity<{ width: number }>`
  padding: ${({ theme }) => theme.spacing(1)}px 0px;
  margin-bottom: ${({ theme }) => theme.spacing(1)}px;
  align-items: center;
  width: ${(props) => props.width}px;
`;

const IconContainer = styled.View<{ backgroundColor: string }>`
  background-color: ${(props) => props.backgroundColor};
  padding: ${({ theme }) => theme.spacing(1)}px;
  justify-content: center;
  align-items: center;
  border-radius: ${({ theme }) => theme.borderRadius.round}px;
  aspect-ratio: 1;
`;

const Label = styled.Text`
  color: ${({ theme }) => theme.palette.text.secondary};
  font-size: 11px;
  text-align: center;
`;

interface MenuItemProps {
  label: string;
  iconName: string;
  backgroundColor?: string;
  onPress: () => any;
  width: number;
}

const MenuItem = (props: MenuItemProps) => {
  const { label, iconName, backgroundColor, width, onPress } = props;
  return (
    <Container width={width} onPress={onPress}>
      <IconContainer backgroundColor={backgroundColor ?? myTheme.palette.menu.default}>
        <Icon
          size={32}
          name={iconName}
          color={backgroundColor ? '#FFFFFF' : '#555'}
        />
      </IconContainer>
      <Label>{label}</Label>
    </Container>
  );
};

export default MenuItem;
