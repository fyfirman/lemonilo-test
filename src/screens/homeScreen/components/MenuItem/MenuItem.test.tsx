import React from 'react';
import { renderWithTheme } from '../../../../__tests__/helpers';
import MenuItem from './MenuItem';
import 'jest-styled-components';

test('should match with snapshot', () => {
  const data = { label: 'Flight', iconName: 'airplane', color: '#2FC6F3' };

  const tree = renderWithTheme(
    <MenuItem iconName={data.iconName} label={data.label} onPress={jest.fn} width={60} />,
  );

  expect(tree).toMatchSnapshot();
});
