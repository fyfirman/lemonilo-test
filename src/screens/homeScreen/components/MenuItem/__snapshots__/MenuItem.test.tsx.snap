// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match with snapshot 1`] = `
<View
  accessible={true}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "marginBottom": 8,
      "opacity": 1,
      "paddingBottom": 8,
      "paddingLeft": 0,
      "paddingRight": 0,
      "paddingTop": 8,
      "width": 60,
    }
  }
>
  <View
    backgroundColor="#F2F3EE"
    style={
      Array [
        Object {
          "alignItems": "center",
          "aspectRatio": 1,
          "backgroundColor": "#F2F3EE",
          "borderRadius": 1000,
          "justifyContent": "center",
          "paddingBottom": 8,
          "paddingLeft": 8,
          "paddingRight": 8,
          "paddingTop": 8,
        },
      ]
    }
  >
    <Text
      allowFontScaling={false}
      style={
        Array [
          Object {
            "color": "#555",
            "fontSize": 32,
          },
          undefined,
          Object {
            "fontFamily": "Ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
        ]
      }
    >
      ï„·
    </Text>
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "rgba(0,0,0,0.5)",
          "fontSize": 11,
          "textAlign": "center",
        },
      ]
    }
  >
    Flight
  </Text>
</View>
`;
